src/main/java/net/dv8tion/jda/api/hooks/AnnotatedEventManager.java

```diff
@@ -114,31 +114,36 @@ private void updateMethods()
         methods.clear();
         for (Object listener : listeners)
         {
-            boolean isClass = listener instanceof Class;
-            Class<?> c = isClass ? (Class) listener : listener.getClass();
-            Method[] allMethods = c.getDeclaredMethods();
-            for (Method m : allMethods)
+            addMethod(listener);
+        }
+    }
+
+    private void addMethod(@Nonnull Object listener)
```

Comment:
Lets rename this `scanListenerForMethods` or something similar as `addMethod` doesn't really do it for me.